src/ui/public/kbn_top_nav/kbn_top_nav_controller.js

```diff
@@ -53,13 +53,18 @@ export default function ($compile) {
 
     // apply the defaults to individual options
     _applyOptDefault(opt = {}) {
-      return defaults({}, opt, {
+      const defaultedOpt = defaults({}, opt, {
         label: capitalize(opt.key),
         hasFunction: !!opt.run,
         description: opt.run ? opt.key : `Toggle ${opt.key} view`,
-        hideButton: !!opt.hideButton,
-        run: (item) => this.toggle(item.key)
+        run: (item) => !item.disableButton && this.toggle(item.key)
       });
+
+      defaultedOpt.hideButton = result(opt, 'hideButton', false);
+      defaultedOpt.disableButton = result(opt, 'disableButton', false);
+      defaultedOpt.tooltip = result(opt, 'tooltip', '');
```

Comment:
`result` accepts strings and functions... do we want to be so permissive here? Or should we instead expect `hideButton`, `disableButton`, and `tooltip` to always be functions, e.g.

```javascript
defaultedOpt.hideButton = opt.hideButton ? opt.hideButton() : false;
```